h/storage.py

```diff
@@ -184,7 +176,7 @@ def update_annotation(request, id_, data, group_service):
     annotation = request.db.query(models.Annotation).get(id_)
     annotation.updated = updated
 
-    group = group_service.find(annotation.groupid)
+    group = request.find_service(IGroupService).find(annotation.groupid)
```

Comment:
annotation.group here too?

Reply:
Yeah unfortunately this is was the source of loads of pain